app/src/main/java/com/oracle/javafx/scenebuilder/app/SceneBuilderApp.java

```diff
@@ -510,21 +529,76 @@ public void handleOpenFilesAction(List<String> files) {
 
         // Fix for #45
         if (userLibrary.isFirstExplorationCompleted()) {
-            performOpenFiles(fileObjs);
+            var openResult = performOpenFiles(fileObjs);
+            handleFileOpenResult(openResult, onSuccess);
         } else {
             // open files only after the first exploration has finished
             userLibrary.firstExplorationCompletedProperty().addListener(new InvalidationListener() {
                 @Override
                 public void invalidated(Observable observable) {
                     if (userLibrary.isFirstExplorationCompleted()) {
-                        performOpenFiles(fileObjs);
+                        var openResult = performOpenFiles(fileObjs);
                         userLibrary.firstExplorationCompletedProperty().removeListener(this);
+                        handleFileOpenResult(openResult, onSuccess);
                     }
                 }
             });
         }
     }
 
+    /**
+     * The onSuccess Runnable is only executed if the {@link FileOpenResult} has no
+     * errors at all.
+     * 
+     * @param result    {@link FileOpenResult} holding the list of files to be
+     *                  opened and a map of files with exceptions in case of an
+     *                  error.
+     * @param onSuccess A runnable holding the action to be performed on success.
+     */
+    private void handleFileOpenResult(FileOpenResult result, Runnable onSuccess) {
+        if (result.hasErrors()) {
+            showFileOpenErrors(result);
+        } else {
+            if (onSuccess != null) {
```

Comment:
As per above comment, if we treat all parameters as non-null, then the null check is not needed.

Reply:
Agree, thats now implemented. Very good proposal, thank.